### PR TITLE
ISO-8859-1 encoding

### PR DESCRIPTION
**Pull request by chroniclekevinpowe**
Added test ensuring all printable characters converted correctly for ISO-8859-1 encoding. (FIX standard's default single byte encoding scheme (https://www.fixtrading.org/standards/tagvalue-online/#character-encoding)

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/bytes/Bytes.java [hunk 1]
```diff
@@ -635,19 +635,22 @@ static String toString(@NotNull final Bytes<?> buffer,
...
+     *
+     * <p>This method supports all characters in the Basic Latin Unicode block.</p>
      *
      * @param buffer   The buffer to extract the string from. Must not be {@code null}.
      * @param position The position in the buffer to start extracting the string from.
      * @param length   The number of characters to include in the extracted string.
      * @return A string extracted from the buffer.
      * @throws IllegalArgumentException If the provided {@code position} or {@code length} is negative.
      * @throws NullPointerException     If the provided {@code buffer} is {@code null}.
+     * @see <a href="https://en.wikipedia.org/wiki/Basic_Latin_(Unicode_block)" target="_blank">Basic Latin Unicode block</a>
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/bytes/Bytes.java:616]: love it

File: src/test/java/net/openhft/chronicle/bytes/ISO88591CharsTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            (byte) 0xC0, (byte) 0xC1, (byte) 0xC2, (byte) 0xC3, (byte) 0xC4, (byte) 0xC5, (byte) 0xC6, (byte) 0xC7, (byte) 0xC8, (byte) 0xC9, (byte) 0xCA, (byte) 0xCB, (byte) 0xCC, (byte) 0xCD, (byte) 0xCE, (byte) 0xCF,
+            (byte) 0xD0, (byte) 0xD1, (byte) 0xD2, (byte) 0xD3, (byte) 0xD4, (byte) 0xD5, (byte) 0xD6, (byte) 0xD7, (byte) 0xD8, (byte) 0xD9, (byte) 0xDA, (byte) 0xDB, (byte) 0xDC, (byte) 0xDD, (byte) 0xDE, (byte) 0xDF,
+            (byte) 0xE0, (byte) 0xE1, (byte) 0xE2, (byte) 0xE3, (byte) 0xE4, (byte) 0xE5, (byte) 0xE6, (byte) 0xE7, (byte) 0xE8, (byte) 0xE9, (byte) 0xEA, (byte) 0xEB, (byte) 0xEC, (byte) 0xED, (byte) 0xEE, (byte) 0xEF,
+            (byte) 0xF0, (byte) 0xF1, (byte) 0xF2, (byte) 0xF3, (byte) 0xF4, (byte) 0xF5, (byte) 0xF6, (byte) 0xF7, (byte) 0xF8, (byte) 0xF9, (byte) 0xFA, (byte) 0xFB, (byte) 0xFC, (byte) 0xFD, (byte) 0xFE, (byte) 0xFF,
+    };
+
+    @Test
+    public void testBytesToStringConversion() {
+        Bytes<?> isoBytes = Bytes.wrapForRead(ISO_BYTES);
+        assertEquals(ISO_TEXT, BytesInternal.to8bitString(isoBytes));
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/bytes/ISO88591CharsTest.java]: pls expose this method publicly via `Bytes`

File: src/main/java/net/openhft/chronicle/bytes/Bytes.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/bytes/Bytes.java:616]: please mention that it won't handle the upper block and refer to Bytes.to8BitString

Comment (chroniclekevinpowe) [reply src/main/java/net/openhft/chronicle/bytes/Bytes.java:616]: Done!

File: src/test/java/net/openhft/chronicle/bytes/ISO88591CharsTest.java (hunk 2 above)
Comment (chroniclekevinpowe) [reply src/test/java/net/openhft/chronicle/bytes/ISO88591CharsTest.java]: Done!

File: src/main/java/net/openhft/chronicle/bytes/Bytes.java [hunk 3]
```diff
@@ -593,61 +593,27 @@ static String toString(@NotNull final Bytes<?> buffer)
     static String toString(@NotNull final Bytes<?> buffer,
                            @NonNegative final long maxLen)
             throws BufferUnderflowException, IllegalArgumentException {
-        requireNonNegative(maxLen);
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/bytes/Bytes.java:596]: dedupe

**Review by JerryShea — APPROVED** [review]

**Pull request merged**